docs(ux-writing): Adds conversation design and chatbot guidelines. (#4206)
patternfly/patternfly-org: packages/documentation-site/patternfly-docs/content/PatternFly-AI/conversation-design/conversation-design.md

---
id: Conversation design

> section: PatternFly-AI

Reviewer: Which sections do we want this and the chatbots.md files to appear in the nav? Should they be under patternfly-ai/conversation-design as different tabs?

if so, i think you should be able to do that by using 
```
id: (page title)
section: PatternFly-AI
source: (page title again? ... really it's whatever you want to be on the tab)
```

---

import { Button, Flex, FlexItem } from '@patternfly/react-core';
import ArrowRightIcon from '@patternfly/react-icons/dist/esm/icons/arrow-right-icon';

# Conversation design guidelines

**Conversation design** is method of writing for conversational interfaces, like chatbots or voicebots. The goal of conversation design is to create an interactive experience that resembles human-to-human conversation as much as possible. Like traditional content design, conversation design is focused on using words to make experiences clear, concise, and well-timed.

Good conversation design combines content strategy, writing, and design to ensure that the development and use of chatbots is centered on real user needs. This includes making sure that all text related to the conversational UI supports easy engagement with chatbots, while staying aligned with brand standards.

> When designing AI conversations alongside PatternFly-based projects, it is important to align with our established [brand voice and tone](/ux-writing/brand-voice-and-tone), as well as our [ethical guidelines for AI.](/patternfly-ai/ai-guidelines)

Reviewer 2: "ethical guidelines for AI" link is broken.

Author: this page should be added by https://github.com/patternfly/patternfly-org/pull/4175! will probably be worth a quick double check on links once all the AI docs are merged to make sure they work

## Best practices 

Following these best practices to help ensure that your users can complete their goals through an AI-based conversation: 

- Be transparent about AI use. 
- Be direct, brief, and consistent. 
- Frame questions in terms of value for the user.
- If you ask for personal info, tell users "why" you're asking first. 
- Always have the last word.

## Writing for chatbots

> When chatbots are design to meet the needs of your users, they can improve the overall UX of your product, because they are convenient, efficient, and persistent. 

Reviewer 2: ```suggestion
When chatbots are designed to meet the needs of your users, they can improve the overall UX of your product. They are convenient, efficient, and persistent. 
```

> [Our virtual assistant extension](/patterfnfly-ai/chatbot/about-virtual-assistant) utilizes PatternFly components to create a foundation for an AI-based chatbot, with additional customization options.

Reviewer 2: ```suggestion
[Our virtual assistant extension](/patternfly-ai/chatbot/about-virtual-assistant) utilizes PatternFly components to create a foundation for an AI-based chatbot, with additional customization options.
```

Reviewer 2: This link doesn't exist either though; I think we are using patternfly-ai/chatbot/about-chatbot?

- Be direct and limit choices for users as much as possible. 
- Unlike traditional content design, *don't lead with the benefit.* End with the benefit to be sure that its read and acted upon. Users should be able to complete an action solely based on a chatbot's last sentence. 
- Frame any questions from the bot in terms of value for the user.
- Provide examples when relevant.
- Avoid open ended questions. If you ask any, make them very simple, increase pause time between messages, and tell users when they should type. 
- Utilize different methods of presenting information, like cards or buttons. You don't have to always use plain text in your messages; sometimes different message types can be more convenient and engaging.

![Chatbot prompt examples](./img/chatbot-prompts.png)

### AI disclosure 

Make sure to disclose any use of AI in chatbots. Our users should be able to trust that we are honest and transparent with them as much as possible.

- Use labels and other visual styling cues to clearly identify AI features. 
- Add necessary legal disclosures where you can, like in the chatbot footer. 

> - Display an indicator when the bot is "thinking" or "typing", so that users know to expect feedback.

Reviewer 2: ```suggestion
- Display an indicator when the bot is "thinking" or "typing," so that users know to expect feedback.
```
